### PR TITLE
Partial dependence for models other than ranger

### PR DESCRIPTION
**Pull request by hideaki**
# Description
Partial dependence for models other than ranger, which are rpart, lm and glm.

# Checklist
Make sure you have performed following items before submitting this pull request.
If not, please describe the reason.  

- [x] Add test cases for this fix/enhancement
- [ ] Pass devtools::check()
- [ ] Pass devtools::test()
- [ ] Test installing from github
- [x] Tested with Exploratory

### CODE REVIEW

File: R/build_lm.R [hunk 1]
```diff
@@ -1,3 +1,51 @@
+
+
+partial_dependence.lm_exploratory = function(fit, target, vars = colnames(data),
```
Comment (hideaki) [inline R/build_lm.R]: Add comment.

File: R/partial_dependence.R [hunk 2]
```diff
@@ -0,0 +1,64 @@
+
+handle_partial_dependence <- function(x) {
```
Comment (hideaki) [inline R/partial_dependence.R]: Add comment.

File: R/randomForest_tidiers.R [hunk 3]
```diff
@@ -2670,6 +2625,53 @@ glance.rpart <- function(x, pretty.name = FALSE, ...) {
   ret
 }
 
+partial_dependence.rpart = function(fit, target, vars = colnames(data),
```
Comment (hideaki) [inline R/randomForest_tidiers.R]: Add comment.

File: R/build_lm.R [hunk 4]
```diff
@@ -1,3 +1,52 @@
+
+
+# Builds partial_dependency object for lm/glm with same structure (a data.frame with attributes.) as edarf::partial_dependence.
+partial_dependence.lm_exploratory = function(fit, target, vars = colnames(data),
```
Comment (kei51e) [inline R/build_lm.R]: Should use <- operator

File: R/build_lm.R [hunk 5]
```diff
@@ -1,3 +1,52 @@
...
+  n = c(min(nrow(unique(data[, vars, drop = FALSE])), 25L), nrow(data)),
```
Comment (kei51e) [inline R/build_lm.R]: Where the 25 comes from?

File: R/build_lm.R [hunk 6]
```diff
@@ -1,3 +1,52 @@
...
+  interaction = FALSE, uniform = TRUE, data, ...) {
+
+  predict.fun = function(object, newdata) {
```
Comment (kei51e) [inline R/build_lm.R]: Should use <- assign operator.

File: R/build_lm.R [hunk 7]
```diff
@@ -1,3 +1,52 @@
...
+    predict(object, newdata = newdata, type = "response")
+  }
+
+  aggregate.fun = function(x) {
```
Comment (kei51e) [inline R/build_lm.R]: Should use <- assign operator.

File: R/build_lm.R [hunk 8]
```diff
@@ -1,3 +1,52 @@
...
+    c("preds" = mean(x))
+  }
+
+  args = list(
```
Comment (kei51e) [inline R/build_lm.R]: Should use <- assign operator.

File: R/build_lm.R [hunk 9]
```diff
@@ -1,3 +1,52 @@
...
+    "vars" = vars,
+    "n" = n,
+    "model" = fit,
+    "uniform" = uniform,
+    "predict.fun" = predict.fun,
+    "aggregate.fun" = aggregate.fun,
+    ...
+  )
+  
+  if (length(vars) > 1L & !interaction) {
```
Comment (kei51e) [inline R/build_lm.R]: Please add some comment for this condition.

File: R/randomForest_tidiers.R [hunk 10]
```diff
@@ -2670,6 +2625,54 @@ glance.rpart <- function(x, pretty.name = FALSE, ...) {
...
+    "data" = data,
+    "vars" = vars,
+    "n" = n,
+    "model" = fit,
+    "uniform" = uniform,
+    "predict.fun" = predict.fun,
+    ...
+  )
+  
+  if (length(vars) > 1L & !interaction) {
```
Comment (kei51e) [inline R/randomForest_tidiers.R]: Please add some comment for this condition.

File: tests/testthat/test_rpart_2.R [hunk 11]
```diff
@@ -50,6 +52,8 @@ test_that("exp_rpart(binary) evaluate training and test", {
   model_df <- flight %>% dplyr::mutate(is_delayed = as.logical(`is delayed`)) %>%
                 exp_rpart(is_delayed, `DIS TANCE`, `DEP DELAY`, test_rate = 0.3, binary_classification_threshold=0.5)
 
+  ret <-  model_df %>% rf_partial_dependence()
```
Comment (kei51e) [inline tests/testthat/test_rpart_2.R]: Does this cover following case?
- data including NA
- data with NA
- data including Inf/-Inf
- data with Inf/-Inf

File: R/build_lm.R (hunk 4 above)
Comment (hideaki) [reply R/build_lm.R]: Changed.

File: R/build_lm.R (hunk 5 above)
Comment (hideaki) [reply R/build_lm.R]: Added comment.

File: R/build_lm.R (hunk 6 above)
Comment (hideaki) [reply R/build_lm.R]: Changed.

File: R/build_lm.R (hunk 7 above)
Comment (hideaki) [reply R/build_lm.R]: Changed.

File: R/build_lm.R (hunk 8 above)
Comment (hideaki) [reply R/build_lm.R]: Changed.

File: R/build_lm.R (hunk 9 above)
Comment (hideaki) [reply R/build_lm.R]: Added comment.

File: R/randomForest_tidiers.R (hunk 10 above)
Comment (hideaki) [reply R/randomForest_tidiers.R]: Added comment.

File: tests/testthat/test_rpart_2.R (hunk 11 above)
Comment (hideaki) [reply tests/testthat/test_rpart_2.R]: Those are handled before building models.

**Pull request merged**